instrumentation/akka/akka-actor-2.3/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/akkaactor/AkkaScheduleInstrumentation.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.akkaactor;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static net.bytebuddy.matcher.ElementMatchers.takesArgument;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.javaagent.bootstrap.Java8BytecodeBridge;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+public class AkkaScheduleInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("akka.actor.LightArrayRevolverScheduler");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        named("schedule")
+            .and(takesArgument(0, named("scala.concurrent.duration.FiniteDuration")))
+            .and(takesArgument(1, named("scala.concurrent.duration.FiniteDuration")))
+            .and(takesArgument(2, named("java.lang.Runnable")))
+            .and(takesArgument(3, named("scala.concurrent.ExecutionContext"))),
+        AkkaScheduleInstrumentation.class.getName() + "$ScheduleAdvice");
+    transformer.applyAdviceToMethod(
+        named("scheduleOnce")
+            .and(takesArgument(0, named("scala.concurrent.duration.FiniteDuration")))
+            .and(takesArgument(1, named("java.lang.Runnable")))
+            .and(takesArgument(2, named("scala.concurrent.ExecutionContext"))),
+        AkkaScheduleInstrumentation.class.getName() + "$ScheduleOnceAdvice");
+  }
+
+  @SuppressWarnings("unused")
+  public static class ScheduleAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void enterSchedule(
+        @Advice.Argument(value = 2, readOnly = false) Runnable runnable) {
+      Context context = Java8BytecodeBridge.currentContext();
```

Comment:
this is just for my own curiosity: why getting the current context from Java8BytecodeBride? 
why not 'Context.current()' directly?

Reply:
This is what most of the code does so I did it too. Are you an regular contributor to this lib and if so, can you explain why I should change?

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b1845878bdde0ac36e6a88c18e31ae2b7da67f81/docs/contributing/javaagent-structure.md?plain=1#L43 says to use Java8BytecodeBridge.currentContext() inside Advice classes.

I assume this includes here but I am a first time contributor so I will change this if any of the regular contributors say so.

Reply:
it's needed when the Advice is inlined into bytecode that targets a version of Java prior to Java 8

interestingly, I think it will no longer be needed after #11457 (since we will no longer be inlining our advice code)

the javadoc explains a bit more:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/b1845878bdde0ac36e6a88c18e31ae2b7da67f81/javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/bootstrap/Java8BytecodeBridge.java#L13-L17

Reply:
> This is what most of the code does so I did it too. Are you a regular contributor to this lib and if so, can you explain why I should change?

i was not asking for a change.. just curious to know the reason behind. @trask explained it.

Reply:
@trask The existing opentelemetry-java-instrumentation code for Akka and Pekko uses Java8BytecodeBridge. Is it ok if I stick this 'style' and that all the code can be changed to stop using Java8BytecodeBridge in a future commit?

Reply:
oh, definitely, it's totally fine to use `Java8BytecodeBridge`